builder/vsphere/common/step_export.go

```diff
@@ -0,0 +1,225 @@
+//go:generate struct-markdown
+//go:generate mapstructure-to-hcl2 -type ExportConfig
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/nfc"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+var sha = map[int]func() hash.Hash{
+	1:   sha1.New,
+	256: sha256.New,
+	512: sha512.New,
+}
+
+type ExportConfig struct {
```

Comment:
I'd like to see a Prepare() method so that we can perform some input validation on the values provided, before we get to exporting at the very end of a potentially long build run.

Reply:
moved this to the prepare